src/Image/Image.php

```diff
@@ -271,14 +270,14 @@ public function setBorderRadius(int $cornerRadius): self
     }
 
     /**
-     * @param float opacity The opacity of the image
+     * @param  ?float  $opacity The opacity of the image
      * @return Image
      *
      * @throws \ImagickException
      */
-    public function setOpacity(float $opacity): self
+    public function setOpacity(?float $opacity = null): self
     {
-        if ((empty($opacity) && $opacity != 0) || $opacity == 1) {
```

Comment:
@lohanidamodar, @eldadfux, I found these related old commits:

* https://github.com/utopia-php/image/pull/3/files#r611330500
* https://github.com/utopia-php/image/pull/2/files#diff-7031b31652730041abe5f4d32a50811dbe847ec7a189b645f7e9b533979fa370

But I don't quite understand the point of this line and what it's really trying to validateor do. I would expect passing 1 is a valid value that should be passed to `$this->image->setImageAlpha($opacity);`. The `empty()` check doesn't quite make sense to me either.

Reply:
I think the empty check was for null value and empty counted 0 as empty hence !=0, may be `is_null` would solve this, not sure why I didn't think of that.

Regarding 1, setting opacity to 1 and giving back same image should be same, hence not setting 1 and just returning as it is to save resource.

Reply:
@stnguyen90 ![Screenshot 2023-04-07 at 2 41 22 AM](https://user-images.githubusercontent.com/20852629/230506895-d70e2ae0-ced3-4e3d-8459-7669044a38d5.png)

Reply:
@christyjacob4, what's the point of allowing `null` input?

Reply:
Looks like even the opacity function can be rewritten as 

public function setOpacity(float $opacity): self
{
    if ($opacity == 1) {
         return $this;
     }
}

Reply:
I have updated the code 👍